Code/Framework/AzCore/Tests/AZStd/String.cpp

```diff
@@ -2470,7 +2470,7 @@ namespace UnitTest
         // Is the size of the pointer (used for storing the memory address of the string)
         // + the size of the string "size" member used to store the size of the string
         // + the size of the string "capacity" member used to store the capacity of the string
-        size_t constexpr ExpectedBasicStringSize = sizeof(void*) + 2 * sizeof(size_t) + sizeof(AZStd::allocator);
+        size_t constexpr ExpectedBasicStringSize = sizeof(void*) + 2 * sizeof(size_t);
```

Comment:
You have reduced the `AZStd::allocator` down to an empty class, therefore it counts for zero bytes when the empty base optimization is being used?
Good job.

Reply:
You can update the static_assert comment to mention that when the basic_string class is used with a stateless_allocator it should result in 24 byte class.